src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final ColorSupplier colorSupplier = new ColorSupplier();
+    private static final int maxParameterValue = 50;
+    private static final int maxFigureNumber = 5;
+    private static final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        double randomParameter = new Random().nextInt(maxParameterValue);
+        int value = random.nextInt(maxFigureNumber);
+        switch (value) {
+            case 0:
+                return new Square(colorSupplier.getRandomColor(), randomParameter);
+            case 1:
+                return new Rectangle(colorSupplier.getRandomColor(), randomParameter,

```

Comment:
Also, don't name it `randomParameter`. Create explicit variable `width`, `radius`, `leg` etc.